src/python2/lister_og_indekser/lister_og_indekser.md

```diff
@@ -0,0 +1,252 @@
+
+```python
+>>> lst = ['egg', 'ham', 'spam']
+>>> lst
+['egg', 'ham', 'spam']
+```
+
+Vi har nå en liste som inneholder ordene `'egg'`, `'ham'` og `'spam'`.
+Vanskeligere er det ikke! Vi kan også lage tomme lister:
+
+```python
+>>> lst = []
+>>> lst
+[]
+```
+
+Ei liste kan inneholde alt mulig - tall, strenger og også andre lister:
+
+```python
+>>> lst = [ 3, 'komma', [1 , 4] ]
+>>> lst
+[3, 'komma', [1, 4]]
+```
+
+
+# Legge til og fjerne elementer {.activity}
+
+# {.protip}
+
+Hva om vi ønsker å legge til eller fjerne elementene fra lista vi vår? Vi skal
+lære om to funksjoner for å gjøre dette - `lst.append(elm)` og `lst.remove(elm)`, der
+`lst` er lista og `elm` er elementet vi ønsker å legge til eller fjerne.
+
+I det følgende eksempelet ser vi hvordan 'lst.append(elm)' fungerer:
+
+```python
+>>> lst = []
+>>> lst.append('Per')
+>>> lst
+['Per']
+>>> lst.append('Ada')
+>>> lst.append('Kim')
+>>> lst
+['Per', 'Ada', 'Kim']
+```
+
+Og her ser vi hvordan `lst.remove(elm)` fungerer:
+
+```python
+>>> lst = ['Per', 'Ada', 'Kim']
+>>> lst.remove("Kim")
+>>> lst
+['Per', 'Ada']
+```
+
+# {.check}
+
+Vi har lyst til å skrive et handleliste-program som lar brukeren skrive inn
+matvarer helt til `'skriv ut'` skrives inn. Da skal programmet skrive ut
+gjenstandene i lista. Programmet kan fungere slik:
+
+<pre>
+>>>
+Skriv inn en gjenstand: <font color="green">ost</font>
+Skriv inn en gjenstand: <font color="green">melk</font>
+Skriv inn en gjenstand: <font color="green">brød</font>
+Skriv inn en gjenstand: <font color="green">skriv ut</font>
+ost
+melk
+brød
+</pre>
+
+Dette må du gjøre:
+
+ * Lag ei tom liste
+ * Så lenge brukeren ikke skriver inn `'skriv ut'`, legg til det nye elementet
+   i lista
+
+     **Hint:** Hva slags løkke kan vi bruke her?
+
+ * Skriv ut hvert hvert element i lista
+
+    **Hint:** Hva slags løkke kan vi bruke her?
+
+# Indekser {.activity}
+
+# {.protip}
+
+Tenk deg at vi har ei liste, og ønsker å hente ut det andre elementet i lista.
+Hvordan skal vi klare det?  Da bruker vi indekser. Da bruker vi `[` og `]` med
+et tall mellom. Slik:
+
+```python
+>>> lst = [1, 2, 3, 4, 5]
+>>> lst[1]
+2
+```
+
+Du la kanskje merke til at vi skrev `1`, men fikk ut det andre elementet i
+lista? Det er fordi vi begynner å telle fra `0`. Dermed har det første
+elementet i lista indeks `0`, og det andre har indeks `1`. Datamaskiner
+begynner å telle på null! Du husker kanskje at det samme skjer når vi bruker `range()`?
+
+```python
+>>> list(range(5))
+[0, 1, 2, 3, 4]
+```
+
+Til nå har vi brukt `for element in lst` for å gå igjennom elementene i lista,
+men noen ganger kan det være praktisk å bruke indekser til det samme. Da bruker
+vi `range()` og `len()`. Husk at `len()` returnerer lengden til lista. De to
+følgende kodesnuttene gjør akkurat det samme:
+
+```python
+lst = ['Per', 'Kim', 'Ada']
+for element in lst:
+    print(element)
+```
+
+```python
+lst = ['Per', 'Kim', 'Ada']
+for i in range(len(lst))
```

Comment:
Jeg er enig med Arve. Det generelle konseptet bak løkker i python i er iteratorer og `for element in lst:`. Det er aldri noen grunn til å skrive `for i in range(len(lst)):` i Python, og jeg ser derfor ikke helt hvorfor vi skal lære det bort? Indekstreningen hører nok bedre sammen med slicing enn med for-løkker. :)